dex/src/main/java/io/crate/data/BatchConsumer.java

```diff
@@ -93,4 +94,16 @@ loop as long as moveNext()==true
      */
     @Override
     void accept(BatchIterator iterator, @Nullable Throwable failure);
+
+    /**
+     * @return true if the consumer wants to scroll backwards by using {@link BatchIterator#moveToStart}
+     */
+    default boolean requiresScroll() {
+        return false;
+    }
+
+    default BatchConsumer projected(List<BatchProjector> projectors) {
```

Comment:
Does it make sense for implementations to overwrite this? Seems like this could be a static method